modules/minimint-mint/src/lib.rs

```diff
@@ -401,14 +389,28 @@ impl Mint {
             db,
         }
     }
-}
 
-impl Mint {
+    fn blind_sign(&self, output: Coins<BlindToken>) -> Result<PartialSigResponse, MintError> {
+        Ok(PartialSigResponse(output.map(
+            |amt, msg| -> Result<_, InvalidAmountTierError> {
+                let sec_key = self.sec_key.tier(&amt)?;
+                let blind_signature = sign_blinded_msg(msg.0, *sec_key);
+                Ok((msg.0, blind_signature))
+            },
+        )?))
+    }
+
     fn combine(
         &self,
         partial_sigs: Vec<(PeerId, PartialSigResponse)>,
     ) -> (Result<SigResponse, CombineError>, MintShareErrors) {
         // FIXME: decide on right boundary place for this invariant
```

Comment:
This FIXME belongs below your insert, it relates to the duplicate sig shares per peer.

Reply:
Ah okay, I was unsure what that comment meant.